cmd/data-usage.go

```diff
@@ -59,6 +61,34 @@ func storeDataUsageInBackend(ctx context.Context, objAPI ObjectLayer, dui <-chan
 	}
 }
 
+// loadPrefixUsageFromBackend returns prefix usages found in passed buckets
+//   e.g.:  /testbucket/prefix => 355601334
+func loadPrefixUsageFromBackend(ctx context.Context, objAPI ObjectLayer, bucket string) (map[string]uint64, error) {
+	z, ok := objAPI.(*erasureServerPools)
+	if !ok {
+		return nil, errors.New("prefix usage is not supported")
+	}
+
+	cache := dataUsageCache{}
+
+	m := make(map[string]uint64)
+	for _, pool := range z.serverPools {
+		for _, er := range pool.sets {
+			// Load bucket usage prefixes
+			if err := cache.load(ctx, er, bucket+slashSeparator+dataUsageCacheName); err == nil {
+				if root := cache.find(bucket); root != nil {
+					for id, usageInfo := range cache.flattenChildrens(*root) {
+						prefix := strings.TrimPrefix(id, bucket+slashSeparator)
+						m[prefix] += uint64(usageInfo.Size)
+					}
+				}
+			}
```

Comment:
```suggestion
			if err := cache.load(ctx, er, bucket+slashSeparator+dataUsageCacheName); err != nil {
			        continue
			}
```

Can we do less nesting by continuing for error first and then have `if { for {` ?

Reply:
> Can we do less nesting by continuing for error first and then have `if { for {` ?

Yes, done.